pkg/roachpb/api.proto

```diff
@@ -69,6 +69,9 @@ message ResponseHeader {
     // The spanning operation didn't finish because the key limit was
     // exceeded.
     RESUME_KEY_LIMIT = 1;
+    // The spanning operation didn't finish because the target byte limit was
+    // exceeded.
+    RESUME_BYTE_LIMIT = 2;
```

Comment:
I know we have byte-size limited scans already so I wonder if there's a reason they didn't introduce this already, and/or if it will cause confusion if we introduce it here but don't use it there? /cc @nvanbenschoten

Reply:
That's a good question. We seem to be using the `RESUME_KEY_LIMIT` reason in places where this new `RESUME_BYTE_LIMIT` reason would be more appropriate, like:
https://github.com/cockroachdb/cockroach/blob/20f2a5411914f2a09c77b3d7458970347ffff2f2/pkg/kv/kvclient/kvcoord/dist_sender.go#L1326

My best guess as to why we didn't do this in https://github.com/cockroachdb/cockroach/commit/9e11347a053601609ff7edbcbc1bf8f1103f619b is because we would then have needed to plumb this reason back out of MVCC operations. We don't seem to use the reason for anything important (or am I missing something?), so maybe we didn't think it was worth the effort? I'm not totally sure, @tbg might know better.

Reply:
Yes, I didn't see a strong enough reason to introduce it at the time.

PS: I don't have context on this PR.